core/src/main/java/com/linecorp/armeria/server/http/dynamic/Deserializers.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.http.dynamic;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Provides deserializing functionality for reflection.
+ *
+ * @see Path
+ * @see PathParam
+ * @see DynamicHttpServiceBuilder
+ */
+final class Deserializers {
+
+    private static final Map<Class<?>, Function<String, Object>> FUNCTIONS;
+
+    static {
+        FUNCTIONS = new HashMap<>();
+        FUNCTIONS.put(Byte.TYPE, Byte::parseByte);
+        FUNCTIONS.put(Short.TYPE, Short::parseShort);
+        FUNCTIONS.put(Integer.TYPE, Integer::parseInt);
+        FUNCTIONS.put(Long.TYPE, Long::parseLong);
+        FUNCTIONS.put(Float.TYPE, Float::parseFloat);
+        FUNCTIONS.put(Double.TYPE, Double::parseDouble);
+        FUNCTIONS.put(String.class, s -> s);
+    }
+
+    /**
+     * Deserialize given {@code str} to {@code T} type object. e.g., "42" -> 42.
+     *
+     * @throws IllegalArgumentException if {@code str} can't be deserialized to {@code T} type object.
+     */
+    public static <T> T deserialize(String str, Class<T> clazz) {
+        if (!deserializable(clazz)) {
+            throw new IllegalArgumentException("Class " + clazz.getSimpleName() + "can't be deserialized.");
+        }
+        Function<String, Object> function = FUNCTIONS.get(clazz);
+        try {
+            return (T) function.apply(str);
+        } catch (Throwable e) {
+            throw new IllegalArgumentException(
+                    String.format("Can't deserialize %s to type %s", str, clazz.getName()));
```

Comment:
After reviewing the usages from the project, I concluded that `Class#getSimpleName()` is more common in Exception messages than `Class#getName()`. But It does not seem there is some rule - some exception messages (about three only) are using `Class#getName()`.

I think using `Class#getSimpleName()` would be better for the exception messages. If you agree, I will fix the exception messages that is using `Class#getName()`.